core/shared/src/main/scala/rapture/core/modes.scala

```diff
@@ -38,7 +38,7 @@ object Mode extends Mode_1 {
     msg = "No implicit mode was available for $"+"{Group} methods. " +
         "Please import a member of rapture.core.modes, e.g. modes.throwExceptions.")
 trait Mode[+Group <: MethodConstraint] { mode =>
-  type Wrap[+ _, _ <: Exception]
+  type Wrap[_, _ <: Exception]
```

Comment:
I shouldn't remove covariance :) 
Scala 2.12 fixed https://issues.scala-lang.org/browse/SI-8079 (https://github.com/scala/scala/pull/5280). Due to this changes in variance checking, `ExponentialBackoffMode`, `ThrowExceptionsMode`, `Repl` and `KeepCalmAndCarryOnMode` modes wouldn't compile. 

```
covariant type Wrap occurs in contravariant position in type => Repl.this.Wrap[Return, _ <: Exception] of value value
    def unwrap[Return](value: => Wrap[Return, _ <: Exception]): Return = value
```

I believe it's because, in these modes, definition of Wrap is ` type Wrap[+T, E <: Exception] = T`, so T is not covariant.

I guess we can fix this issue by annotating `unwrap` method's argument (only these 4 modes) by `@uncheckedVariance`.
Example: 
`def unwrap[Return](value: => Wrap[Return, _ <: Exception] @uncheckedVariance): Return = value`

But after that, I got an error: 
``` 
unbound wildcard type  def unwrap[Return](value: => Wrap[Return, _ <: Exception] @uncheckedVariance): Return = value
```

To fix this issue I've just replaced unbounded Exception wildcard type for parent `Mode` class.
` def unwrap[Res, E <: Exception](value: => Wrap[Res, E]): Res`

Not sure that's a good idea. Please validate it ;)